include/deal.II/dofs/dof_handler.h

```diff
@@ -472,6 +472,25 @@ class DoFHandler : public Subscriptor
   initialize(const Triangulation<dim, spacedim> &tria,
              const FiniteElement<dim, spacedim> &fe);
 
+  /**
+   * Assign a FiniteElement @p fe to this object.
+   *
+   * @note This function makes a copy of the finite element given as
+   * argument, and stores it as a member variable. Consequently, it is
+   * possible to write code such as
+   * @code
+   *   dof_handler.set_fe(FE_Q<dim>(2));
+   * @endcode
+   * You can then access the finite element later on by calling
+   * DoFHandler::get_fe(). However, it is often more convenient to
+   * keep a named finite element object as a member variable in your
+   * main class and refer to it directly whenever you need to access
+   * properties of the finite element (such as
+   * FiniteElementData::dofs_per_cell). This is what all tutorial programs do.
+   */
+  virtual void
+  set_fe(const FiniteElement<dim, spacedim> &fe);
```

Comment:
I think it would be useful to state that this will generally leave the object in an inconsistent state: the FE is already set to something, but DoF indices are either not distributed or are still distributed to something that uses the previously set FE. You may also want to point to `distribute_dofs()` as the function that restores the object to a consistent state.

Reply:
What do you think about a `@warning`?

Reply:
Sounds good.